src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -56,11 +56,46 @@ public class HubEndPoint<THub, TClient> : EndPoint, IInvocationBinder where THub
         {
             // TODO: Dispatch from the caller
             await Task.Yield();
-            Exception exception = null;
+
             try
             {
                 await _lifetimeManager.OnConnectedAsync(connection);
+                await RunHubAsync(connection);
+            }
+            catch (Exception ex)
```

Comment:
Actually it is needed otherwise we will just close a pending poll with 204 in case of exceptions. As discussed I moved it to the endpoint.